### PR TITLE
Init sanity unit test

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request
https://github.com/tgstation/TerraGov-Marine-Corps/pull/5326
Stemming from 
https://github.com/ParadiseSS13/Paradise/pull/14770

Basically it just checks for bad initialize calls

### CODE REVIEW

File: code/modules/vending/cola.dm [hunk 1]
```diff
@@ -39,7 +39,9 @@
 /obj/machinery/vending/cola/random/Initialize()
 	// No need to call parent, we're not doing anything with this machine. Just picking a new type of machine to use, spawning it and deleting ourselves.
```
Comment (Cyberboss) [inline code/modules/vending/cola.dm:40]: Okay, I don't know who added this, but can we undo it? It's just circumventing 1 proc call of sanity.

This and the snack machine

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — DISMISSED** [review]

**Review by Mothblocks — DISMISSED** [review]

**Comment by Timberpoes** [comment]
I removed the parent calls on random snack machines for a reason. They would spawn in a new machine just to spawn in a new machine and qdel themselves and it was causing some weird issue at the time.

The solution is not to call parent but to turn it into an object spawner similar to random arcades.

Don't just lazily add parent proc calls and use git blame to find the idiot responsible for it in the first place if you're not sure why something was done.

**Review by Timberpoes — CHANGES_REQUESTED** [review]
https://github.com/TiviPlus/tgstation/pull/1

This is the preferred solution to not calling the parent proc which I've taken the liberty to PR to your branch.

Random vending machines repathed to /obj/effect/spawner subtypes and maps updated accordingly using VSC find-and-replace across all DMM files.

**Comment by MrStonedOne** [comment]
You know what I love. Having to follow two fucking cross links to know what a pr does.

**Comment by Cyberboss** [comment]
Yeah, update the desc

**Review by Timberpoes — DISMISSED** [review]
I've already spoken to @TiviPlus about the failed test case.

Issue are snowflake vendors that are mapping var edited to start with contraband enabled. I have suggested Tivi create a `/contraband` subtype of the two random snack/cola vend spawners that sets `extended_inventory = TRUE` for the vending machine created and then replace the two snowflake machines with it, which will resolve this problem.

**Review by Cyberboss — APPROVED** [review]

File: code/modules/unit_tests/initialize_sanity.dm [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/datum/unit_test/initialize_sanity/Run()
+	if(length(SSatoms.BadInitializeCalls))
+		Fail("Bad Initialize() calls detected. Please read logs.")
```
Comment (optimumtact) [inline code/modules/unit_tests/initialize_sanity.dm:3]: you cannot see the logs in CI runs, this log_world calls below should be to_chat(world) probably

Comment (optimumtact) [reply code/modules/unit_tests/initialize_sanity.dm:3]: Or, it should call Fail repeatedly, once for each type of init failure and the atoms responsible

Comment (optimumtact) [reply code/modules/unit_tests/initialize_sanity.dm:3]: @Cyberboss @TiviPlus please change it to use Fail repeatedly

if that isn't quite suitable, then use log_test(text)

Comment (MrStonedOne) [reply code/modules/unit_tests/initialize_sanity.dm:3]: log world goes to stderr which you *can* see in CI.

initialize x subsystem in x seconds is a log_world and you can see those.

Comment (optimumtact) [reply code/modules/unit_tests/initialize_sanity.dm:3]: this printed nothing to the CI and log_test does both a send_text and log call

**Pull request merged**